servers/src/test/java/tachyon/master/file/FileSystemMasterPermissionCheckTest.java

```diff
@@ -0,0 +1,394 @@
+    verifyMkdir(TEST_USER_1, "/dir1", false);
+
+    // mkdir "/dir_admin" for superuser
+    verifyMkdir(TEST_USER_ADMIN, "/dir_admin", false);
+
+    // mkdir "/dir_supergroup" for user in supergroup
+    verifyMkdir(TEST_USER_SUPERGROUP, "/dir_supergroup", false);
+
+    // mkdir nested "/notExistDir/notExistSubDir" for user1
+    verifyMkdir(TEST_USER_1, "/notExistDir/notExistSubDir", true);
+    FileInfo fileInfo = mFileSystemMaster.getFileInfo(mFileSystemMaster.getFileId(
+        new TachyonURI("/notExistDir")));
+    Assert.assertEquals("notExistDir", fileInfo.getName());
+    Assert.assertEquals(true, fileInfo.isFolder);
+    Assert.assertEquals(TEST_USER_1.getUser(), fileInfo.getUserName());
+  }
+
+  @Test
+  public void mkdirSuccessTest() throws Exception {
+    // mkdir "/testDir/dir1" for user1
+    verifyMkdir(TEST_USER_1, TEST_DIR_URI + "/dir1", false);
+  }
+
+  @Test
+  public void mkdirFailTest() throws Exception {
+    mThrown.expect(AccessControlException.class);
+
+    // mkdir "/testDir/dir1" for user2
+    verifyMkdir(TEST_USER_2, TEST_DIR_URI + "/dir1", false);
+  }
+
+  private void verifyMkdir(TestUser user, String path, boolean recursive) throws Exception {
+    PlainSaslServer.AuthorizedClientUser.set(user.getUser());
+    if (recursive) {
+      mFileSystemMaster.mkdir(new TachyonURI(path),
+          new MkdirOptions.Builder(MasterContext.getConf()).setRecursive(true).build());
+    } else {
+      mFileSystemMaster.mkdir(new TachyonURI(path), MkdirOptions.defaults());
+    }
+
+    FileInfo fileInfo = mFileSystemMaster.getFileInfo(mFileSystemMaster.getFileId(
+        new TachyonURI(path)));
+    String[] pathComponents = path.split("/");
+    Assert.assertEquals(pathComponents[pathComponents.length - 1], fileInfo.getName());
+    Assert.assertEquals(true, fileInfo.isFolder);
+    Assert.assertEquals(user.getUser(), fileInfo.getUserName());
+  }
+
+  @Test
+  public void renameUnderRootSuccessTest() throws Exception {
+    // rename "/testFile" to "/testFileRenamed" for user2
+    verifyRename(TEST_USER_2, TEST_FILE_URI, "/testFileRenamed");
+
+    // rename "/testFile" to "/testFileRenamed" for superuser
+    verifyRename(TEST_USER_ADMIN, "/testFileRenamed", TEST_FILE_URI);
+
+    // rename "/testFile" to "/testFileRenamed" for user in supergroup
+    verifyRename(TEST_USER_SUPERGROUP, TEST_FILE_URI, "/testFileRenamed");
+  }
+
+  @Test
+  public void renameUnderRootFailTest() throws Exception {
+    mThrown.expect(AccessControlException.class);
+
+    // rename "/testFile" to "/testFileRenamed" for user1
+    verifyRename(TEST_USER_1, TEST_FILE_URI, "/testFileRenamed");
+  }
+
+  @Test
+  public void renameSuccessTest() throws Exception {
+    // rename "/testDir/file" to "/testDir/fileRenamed" for user1
+    verifyRename(TEST_USER_1, TEST_DIR_FILE_URI, "/testDir/fileRenamed");
+  }
+
+  @Test
+  public void renameFailNotByPermissionTest() throws Exception {
+    mThrown.expect(InvalidPathException.class);
```

Comment:
Could you also verify the expected message for this test and others in this class?